caravel/assets/javascripts/explorev2/components/ControlPanelsContainer.jsx

```diff
@@ -1,20 +1,38 @@
 import React from 'react';
+import { connect } from 'react-redux';
 import { Panel } from 'react-bootstrap';
-import TimeFilter from './TimeFilter';
-import ChartControl from './ChartControl';
-import GroupBy from './GroupBy';
-import SqlClause from './SqlClause';
-import Filters from './Filters';
+import { DefaultControls, VIZ_CONTROL_MAPPING } from '../constants';
 
-const ControlPanelsContainer = function () {
-  return (
-    <Panel>
-      <ChartControl />
-      <TimeFilter />
-      <GroupBy />
-      <SqlClause />
-      <Filters />
-    </Panel>
-  );
+const propTypes = {
+  vizType: React.PropTypes.string,
 };
-export default ControlPanelsContainer;
+
+const defaultProps = {
+  vizType: null,
+};
+
+class ControlPanelsContainer extends React.Component {
```

Comment:
typically if you write a pure function you do so like:
`function ControlPanelsContainer(props) { ... }`

and so you access `props` the same way you would acces `this.props` if it were a class.